extra/training.py

```diff
@@ -34,7 +24,7 @@ def train(model, X_train, Y_train, optim, steps, BS=128, lossfn=sparse_categoric
     # printing
     if not noloss:
       cat = np.argmax(out.numpy(), axis=-1)
-      accuracy = (cat == y).mean()
+      accuracy = (cat == y.numpy()).mean()
```

Comment:
For this part, I have PR that uses cat.argmax() with Tensor.argmax. Maybe you dont need to convert y to numpy.

cat = np.argmax(out.numpy(), axis=-1)
accuracy = (cat == y.numpy()).mean()

if cat is not needed to be numpy than you can do.

cat = out.argmax(-1)
accuracy = (cat == y).mean().numpy()

Reply:
Oh yea, it should absolutely be this. I'll merge this, can fix in the argmax PR